log/stat/stat.go

```diff
@@ -0,0 +1,97 @@
+package stat
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/peak/s5cmd/strutil"
+)
+
+const (
+	totalCount = iota
+	succCount
+	timeInfo
+)
+
+var (
+	enabled bool
+	stats   statistics
+)
+
+type statistics [3]syncMap
+
+// InitStat initializes collecting program statistics.
+func InitStat() {
+	enabled = true
+	for i := range stats {
+		stats[i] = syncMap{
+			mu:          sync.Mutex{},
+			mapStrInt64: map[string]int64{},
+		}
+	}
+}
+
+// syncMap is a statically typed and synchronized map.
+type syncMap struct {
```

Comment:
Let's embed this `mutex` into syncMap
```go
type syncMap struct {
   sync.Mutex
   mapStrInt64 map[string]int64
}
```
and update `add` method to use `lock, defer unlock` pattern

```go
func (s *syncMap) add(key string, val int64) {
	s.Lock()
	defer s.Unlock()

	s.mapStrInt64[key] += val
}
```